hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacitySchedulerConfiguration.java

```diff
@@ -527,10 +527,10 @@ private void throwExceptionForUnexpectedWeight(float weight, String queue,
     }
   }
 
-  public float getNonLabeledQueueWeight(String queue) {
-    String configuredValue = get(getQueuePrefix(queue) + CAPACITY);
+  public float getNonLabeledQueueWeight(QueuePath queue) {
+    String configuredValue = get(getQueuePrefix(queue.getFullPath()) + CAPACITY);
```

Comment:
There are many string operations in this class: 
E.g.
- getQueuePrefix that works with the full queue path
- getNodeLabelPrefix that also works with the full queue path

I suggest to create a static class, called "QueuePrefixes" or something like that and add some static methods there to convert the QueuePath object to those various queue prefix strings that are ultimately keys in the Configuration object.

Reply:
A followup Jira is created: https://issues.apache.org/jira/browse/YARN-11041